VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -15,20 +24,41 @@
     private DiscussionTextUtils() {
     }
 
-    public static CharSequence getAuthorAttributionText(@NonNull final IAuthorData authorData, @NonNull final Resources resources) {
-        final CharSequence formattedTime = DateUtils.getRelativeTimeSpanString(
-                authorData.getCreatedAt().getTime(),
-                System.currentTimeMillis(),
-                DateUtils.MINUTE_IN_MILLIS,
-                DateUtils.FORMAT_NUMERIC_DATE|DateUtils.FORMAT_SHOW_YEAR);
+    public static void setAuthorAttributionText(@NonNull TextView textView, @NonNull final IAuthorData authorData, @NonNull final Router router) {
+        final CharSequence text;
+        {
+            final CharSequence formattedTime = DateUtils.getRelativeTimeSpanString(
+                    authorData.getCreatedAt().getTime(),
+                    System.currentTimeMillis(),
+                    DateUtils.MINUTE_IN_MILLIS,
+                    DateUtils.FORMAT_NUMERIC_DATE | DateUtils.FORMAT_SHOW_YEAR);
 
-        final String authorLabel = authorData.getAuthorLabel();
+            final String authorLabel = authorData.getAuthorLabel();
 
-        return trim(ResourceUtil.getFormattedString(resources, R.string.post_attribution, new HashMap<String, CharSequence>() {{
-            put("time", formattedTime);
-            put("author", authorData.getAuthor());
-            put("author_label", null == authorLabel ? "" : authorLabel.toUpperCase(resources.getConfiguration().locale));
-        }}));
+            final Context context = textView.getContext();
+            final SpannableString authorSpan = new SpannableString(authorData.getAuthor());
+            authorSpan.setSpan(new ClickableSpan() {
+                @Override
+                public void onClick(View widget) {
+                    router.showUserProfile(context, authorData.getAuthor());
+                }
+
+                @Override
+                public void updateDrawState(TextPaint ds) {
+                    ds.setColor(context.getResources().getColor(R.color.edx_brand_primary_base));
```

Comment:
Hmm, this should probably be set as the `textColorLink` attribute globally for the `TextView` style I guess?